source/draw/RasterMonochrome.ooc

```diff
@@ -155,6 +155,35 @@ RasterMonochrome: class extends RasterPacked {
 			source += sourceWidth
 		}
 	}
+	getScharrDerivative: func(region: IntBox2D, imageX, imageY: FloatImage) {
+		step := 2
+		sourceWidth := this size width
+
+		_pointer := this buffer pointer + region leftTop y * sourceWidth  // this getValue [x,y]
+		_pointerP1 := _pointer + sourceWidth * step
+		_pointerM1 := _pointer - sourceWidth * step
+		destinationX := imageX pointer
+		destinationY := imageY pointer
+
+		for (y in (region leftTop y)..(region rightBottom y)) {
+			for (x in (region leftTop x)..(region rightBottom x)) {
+
+				rightUp := _pointerM1[x + step]
+				rightDown := _pointerP1[x + step]
+				leftUp := _pointerM1[x - step]
+				leftDown := _pointerP1[x - step]
+
+				destinationX@ = ((1.0f * (_pointer[x + step] - _pointer[x - step]) * 0.4641f) + (1.0f * ((rightUp + rightDown) - (leftUp + leftDown)) * 0.26795f))
```

Comment:
Should this be in Kean or in or own code base?

Den 2015-06-02 16:26, ziquanyu skrev:
>
> In source/draw/RasterMonochrome.ooc 
> <https://github.com/cogneco/ooc-kean/pull/251#discussion_r31527885>:
>
> > +
> > +		_pointer := this buffer pointer + region leftTop y * sourceWidth  // this getValue [x,y]
> > +		_pointerP1 := _pointer + sourceWidth * step
> > +		_pointerM1 := _pointer - sourceWidth * step
> > +		destinationX := imageX pointer
> > +		destinationY := imageY pointer
> > +
> > +		for (y in (region leftTop y)..(region rightBottom y)) {
> > +			for (x in (region leftTop x)..(region rightBottom x)) {
> > +
> > +				rightUp := _pointerM1[x + step]
> > +				rightDown := _pointerP1[x + step]
> > +				leftUp := _pointerM1[x - step]
> > +				leftDown := _pointerP1[x - step]
> > +
> > +				destinationX@ = ((1.0f * (_pointer[x + step] - _pointer[x - step]) * 0.4641f) + (1.0f * ((rightUp + rightDown) - (leftUp + leftDown)) * 0.26795f))
>
> i was thinking to avoid using as Float. but i think it's better to 
> delete it and replace 1.0f with the float number.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/cogneco/ooc-kean/pull/251/files#r31527885>.
>